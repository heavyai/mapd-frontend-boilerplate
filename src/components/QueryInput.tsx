import React from 'react'

import { Dispatcher } from '~/actions'
import { setStatement, sendQuery } from '~/actions/query'
import { QueryState } from '~/reducers/query'

const styles = require( './QueryInput.sass' )


type Props = {
  dispatch: Dispatcher,
  query: QueryState
}

const QueryInput: React.StatelessComponent<Props> = ( { dispatch, query: { statement } } ) => {
  const click = () => {
    dispatch( sendQuery( statement ) )
  }
  const handleInputChange = ( event: React.FormEvent<HTMLInputElement> ) => {
    dispatch( setStatement( event.currentTarget.value ) )
  }

  return (
    <div>
      <input type='text' placeholder='Enter a SQL statement' onChange={ handleInputChange } />
      <button className={ styles.button } onClick={ click }>
        Send
      </button>
    </div>
  )
}


export default QueryInput
