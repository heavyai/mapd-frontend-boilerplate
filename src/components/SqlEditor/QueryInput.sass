.container
  position: relative
  padding: 10px 18px 18px

.input
  width: 100%

.progress
  position: absolute
  height: 3px
  bottom: 25px
