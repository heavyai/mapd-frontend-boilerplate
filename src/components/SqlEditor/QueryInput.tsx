import React from 'react'
import { TextField } from 'rmwc/TextField'
import LinearProgress from 'rmwc/LinearProgress'

import { Dispatcher } from '~/store'
import { setStatement, sendQuery } from '~/actions/query'

const styles = require( './QueryInput.sass' )


type Props = {
  dispatch: Dispatcher,
  statement: string,
  pending: boolean
}

const QueryInput: React.StatelessComponent<Props> = ( { dispatch, statement, pending } ) => {
  const handleInputChange = ( event: React.FormEvent<HTMLInputElement> ) => {
    dispatch( setStatement( event.currentTarget.value ) )
  }

  const handleInputKeyPress = ( event: React.KeyboardEvent<HTMLInputElement> ) => {
    if ( event.key === 'Enter' ) {
      dispatch( sendQuery.action( statement ) )
    }
  }

  return (
    <div className={ styles.container }>
      <TextField
        className={ styles.input }
        label='Enter a SQL statement'
        spellCheck={ false }
        value={ statement }
        onChange={ handleInputChange }
        onKeyPress={ handleInputKeyPress }
      />
      { pending && <LinearProgress className={ styles.progress } determinate={ false } /> }
    </div>
  )
}


export default QueryInput
