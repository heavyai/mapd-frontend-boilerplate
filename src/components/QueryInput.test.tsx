import React from 'react'
import * as renderer from 'react-test-renderer'

import QueryInput from './QueryInput'


const queryState = {
  statement: '',
  request: {
    pending: false
  }
}

test( 'Component renders', () => {
  const component = renderer.create(
    <QueryInput dispatch={ jest.fn() } query={ queryState } />
  )

  expect( component.toJSON() ).toMatchSnapshot()
} )
