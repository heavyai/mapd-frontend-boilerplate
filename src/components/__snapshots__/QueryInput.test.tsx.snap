// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders 1`] = `
<div>
  <input
    onChange={[Function]}
    placeholder="Enter a SQL statement"
    type="text"
  />
  <button
    className="button"
    onClick={[Function]}
  >
    Send
  </button>
</div>
`;
